### ISSUE TITLE
"Real Server"

### ISSUE THREAD

**Issue by demipixel** [FactorioBlueprints/factorio-prints#16]
As to [this comment](https://github.com/FactorioBlueprints/factorio-prints/issues/5#issuecomment-266769902). Essentially, running an actual backend instead of using firebase's.

Reasons to do it:
- Verify blueprint strings and other information (also prevent spamming)
- Could fix issues like #5 
- Will help expansion of new features that could require a backend (read contents of blueprint string?)
- Less load on client (can do more on server)

Reasons not to:
- If server-side rendering was one of the considerations (my personal favorite is EJS...), that would be scraping a lot of the existing react content
- Have to handle logins and databases ourselves. This could mean express+passport+mongo (or whatever combo you choose), but it becomes much more of a hassle (in terms of code, it might be easier to for devs to test since there's no worry about modifying the production database).
- Transfer firebase data (wouldn't be too hard in JSON format, but transferring users is a larger pain since you'd need to transfer the users over, map their firebase-id to their new id, and insert that into the old blueprints when adding those)

TL;DR

Pros: Client runs faster, more control over what we can do, clearer separation between server and client

Cons: Risk scraping a lot of the JS if a lot would be moved to backend, more hassle in terms of databases & logins
